src/App.js

```diff
@@ -13,35 +17,86 @@ Here's what a store item should look like
 
 What should a cart item look like? 🤔
 */
+//const initialStoreItems = []
 
 console.log(initialStoreItems)
 
 export default function App() {
+  const [storeItems, setstoreItems] = useState(initialStoreItems)
+  const [cartItems, setCartItems] = useState([])
   // Setup state here...
+  // add item to cart
+  //create new array that inludes the items
+   //checks if item already exists in cart
+   // if exists returns true,
+    //define quantity
+  const addItemToCart = item => {
+    console.log("push items", item)
+    const exists = cartItems.find(cartItem => cartItem.id === item.id)
+    if (exists) {
+      const updateCartItem = cartItems.map(selectedItem => {
+        if (selectedItem.id === item.id) {
+          return { ...selectedItem, quantity: selectedItem.quantity + 1 }
+        }
+        return selectedItem
+      })
+      setCartItems(updateCartItem)
+    } else {
+      const newCartItem = { ...item, quantity: 1 }
+      setCartItems([...cartItems, newCartItem])
+    }
+  }
+
+  //decrease button
+ 
+  //checks the quantity
+   //if it is more than 1; decrease -1
+  //if its equal to 1; remove it
+  const decreaseButton = cartItem => {
+    if (cartItem.quantity > 1) {
+      const updatedCartItems = cartItems.map(targetCartItem => {
+        if (targetCartItem.name === cartItem.name && targetCartItem.quantity) {
+          return {...targetCartItem, quantity: targetCartItem.quantity - 1}
+        }
+        return targetCartItem
+      })
+      setCartItems(updatedCartItems)
+    } else {
+      const removeFromCart = cartItems.filter( targetCartItem => targetCartItem.name !== cartItem.name)
+   setCartItems(removeFromCart)
+ }
+}
+
+  //increse button
+  const increaseButton = cartItem => {
+     const updatedCartItems = cartItems.map(selectedItem => {
+      if (selectedItem.name === cartItem.name ) {
+        console.log("increase item", selectedItem)
+        return{...selectedItem, quantity: selectedItem.quantity +1}
+      } else {
+        return selectedItem
+      } 
+     })
+     setCartItems(updatedCartItems)
+  }
+//total
+  const cartTotal = () => {
+    let total = 0
+    cartItems.forEach(item => (total += item.quantity * item.price))
+      console.log("total item", cartItems)
+    return `£${total.toFixed(2)}`
+  }
 
+  console.log(cartItems)
```

Comment:
I like how you wrote the main important functions in the main folder so that the child components can all use them as needed